src/main/java/com/example/Restaurant_Management_System_REST_API/model/entity/InventoryItem.java

```diff
@@ -2,20 +2,34 @@
 
 import com.example.Restaurant_Management_System_REST_API.model.CatalogItem;
 import jakarta.persistence.*;
+import jakarta.validation.constraints.Positive;
+import lombok.AllArgsConstructor;
 import lombok.Getter;
+import lombok.NoArgsConstructor;
 import lombok.Setter;
 
 import java.time.LocalDateTime;
 
 @Entity
 @Getter
 @Setter
+@NoArgsConstructor
 public class InventoryItem extends CatalogItem {
+
+    public InventoryItem(Long id, LocalDateTime deliveryDate, int stockAmount, Supplier supplier, String name,
+                         String description, Double price ) {
+        super(name, description, price);
+        this.id = id;
+        this.deliveryDate = deliveryDate;
+        this.stockAmount = stockAmount;
+        this.supplier = supplier;
+    }
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "id", nullable = false)
     private Long id;
     private LocalDateTime deliveryDate;
+    @Positive(message = "Stock amount must be above zero!")
```

Comment:
```suggestion
    @Positive(message = "The stock amount has to be more than zero!")
```

Sounds more English :)

Reply:
my american friend tells me both are correct ;)

Reply:
I'd argue that "above zero" refers to something like depth or height, while we're on amount topic I'd stick to "more/less than" :) Feel free to leave it as you wish

Reply:
so this time it stays as it is ;)
what about the rest ? is everything now OK? can i merge it?

Reply:
I'm not the one to tell you to merge or not to merge :) You decide when to do it